metadata-service/factories/src/main/java/com/linkedin/metadata/boot/UpgradeStep.java

```diff
@@ -0,0 +1,134 @@
+package com.linkedin.metadata.boot;
+
+import com.linkedin.common.AuditStamp;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.data.DataMap;
+import com.linkedin.entity.EntityResponse;
+import com.linkedin.events.metadata.ChangeType;
+import com.linkedin.metadata.Constants;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.key.DataHubUpgradeKey;
+import com.linkedin.metadata.utils.EntityKeyUtils;
+import com.linkedin.metadata.utils.GenericRecordUtils;
+import com.linkedin.mxe.MetadataChangeProposal;
+import com.linkedin.upgrade.DataHubUpgradeRequest;
+import com.linkedin.upgrade.DataHubUpgradeResult;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+
+
+@Slf4j
+public abstract class UpgradeStep implements BootstrapStep {
+  private static final Integer SLEEP_SECONDS = 120;
+
+  protected final EntityService _entityService;
+  private final String version;
+  private final String upgradeId;
+  private final Urn upgradeUrn;
+
+  public UpgradeStep(EntityService entityService, String version, String upgradeId) {
+    this._entityService = entityService;
+    this.version = version;
+    this.upgradeId = upgradeId;
+    this.upgradeUrn = EntityKeyUtils.convertEntityKeyToUrn(new DataHubUpgradeKey().setId(upgradeId),
+        Constants.DATA_HUB_UPGRADE_ENTITY_NAME);
+  }
+
+  @Override
+  public void execute() throws Exception {
+    String upgradeStepName = name();
+
+    log.info(String.format("Attempting to run %s Upgrade Step..", upgradeStepName));
+    log.info(String.format("Waiting %s seconds..", SLEEP_SECONDS));
+
+    // Sleep to ensure deployment process finishes.
+    Thread.sleep(SLEEP_SECONDS * 1000);
+
+    if (hasUpgradeRan()) {
+      return;
+    }
+
+    try {
+      ingestUpgradeRequestAspect();
+      upgrade();
+      ingestUpgradeResultAspect();
+    } catch (Exception e) {
+      String errorMessage = String.format("Error when running the %s Upgrade Step", upgradeStepName);
+      log.error(errorMessage, e);
+      cleanUpgradeAfterError();
+      throw new RuntimeException(errorMessage, e);
+    }
+
+    log.info(String.format("Successfully ran %s Upgrade Step", upgradeStepName));
```

Comment:
Let's print the version as well

Reply:
Maybe even the upgrade urn, version or something

Reply:
changed it to upgradeId and version